### DISCUSSION TITLE
Session expiring immediately on login

### DISCUSSION THREAD

**Discussion by jacklenox** [dani-garcia/vaultwarden#2365] [Q&A]
### Subject of the issue
I am having the exact issue described in #1177. I'm definitely running the latest version of Vaultwarden and have just migrated my data. As far as I can see, the Docker container is running on the same timezone as my computer, and they're both in sync.

### Deployment environment

It's not clear to me how I access the Diagnostics tab without being able to log in. I see a lot of discussion of accessing: `/admin/diagnostics`, I can't seem to access this. I'm trying to reach it via: http://localhost:8008/#/admin/diagnostics

I'm clearly misunderstanding something here.

* vaultwarden version: 1.24.0

* Install method: `sudo docker run -d --name vaultwarden -v ~/.vw-data:/data/ -p 8008:80 vaultwarden/server:latest`

* Clients used:  web vault

* Reverse proxy and version: 

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details:

### Steps to reproduce

Try logging in once server is running.

### Expected behaviour

Successful login.

### Actual behaviour

"Logged out. Your login session has expired."

### Troubleshooting data

```
[2022-03-10 10:58:58.696][request][INFO] POST /api/accounts/prelogin
[2022-03-10 10:58:58.697][response][INFO] POST /api/accounts/prelogin (prelogin) => 200 OK
[2022-03-10 10:58:58.707][request][INFO] POST /identity/connect/token
[2022-03-10 10:58:58.752][vaultwarden::api::identity][INFO] User jacklenox@fastmail.com logged in successfully. IP: 172.17.0.1
[2022-03-10 10:58:58.752][response][INFO] POST /identity/connect/token (login) => 200 OK
[2022-03-10 10:58:58.817][request][INFO] POST /identity/connect/token
[2022-03-10 10:58:58.822][response][INFO] POST /identity/connect/token (login) => 200 OK
[2022-03-10 10:58:58.834][request][INFO] GET /api/sync?excludeDomains=true
[2022-03-10 10:58:58.835][auth][ERROR] Unauthorized Error: Invalid claim
[2022-03-10 10:58:58.835][response][INFO] GET /api/sync?<data..> (sync) => 401 Unauthorized
```

**Comment by BlackDex** [comment]
You should go to http://localhost:8080/admin/diagnostics, so without the `/#/` part, because that is used by the web-vault it self.
Also, I'm not sure how and what you exactly migrated, but if you generated new rsa files which are used for the generation of the JWT, then it probably is an issue with that.

But first try to get the support string.

**Comment by jacklenox** [comment]
Thank you for the very quick response! So, I tried this too, but at this URL I just get the standard:
```
Unable to connect

Firefox can’t establish a connection to the server at localhost:8080.

    The site could be temporarily unavailable or too busy. Try again in a few moments.
    If you are unable to load any pages, check your computer’s network connection.
    If your computer or network is protected by a firewall or proxy, make sure that Firefox is permitted to access the Web.
```

**Comment by jacklenox** [comment]
And to migrate, I just copied my `/bw-data/` directory to my new setup.

**Reply by BlackDex** [reply]
Well, i see i used a wrong port in my example, just change that to the one you are using, which seems to be 8008 instead of mine 8080.

**Reply by jacklenox** [reply]
Agh sorry, yep, I had checked this previously. So when I head to http://localhost:8008/admin/diagnostics I get a 404:

```
404: Not Found

The requested resource could not be found.
Rocket
```

And this in the logs:

```
[2022-03-10 11:22:13.629][start][INFO] Rocket has launched from http://0.0.0.0:80
[2022-03-10 11:22:25.268][request][INFO] GET /admin/diagnostics
[2022-03-10 11:22:25.269][response][INFO] GET /<p..> [10] (web_files) => 404 Not Found
```

**Reply by BlackDex** [reply]
Then i think you have a wrong DOMAIN config, probably with a sub-path or something

**Reply by jacklenox** [reply]
Hmm, the setup I'm using is identical to how it currently runs on my previous machine.

I haven't configured anything, just using default settings.

**Reply by BlackDex** [reply]
I suggest to read the following wiki pages:

- https://github.com/dani-garcia/vaultwarden/wiki/Starting-a-Container
- https://github.com/dani-garcia/vaultwarden/wiki/Using-Docker-Compose
- https://github.com/dani-garcia/vaultwarden/wiki/Enabling-admin-page

**Comment by ranapushpender** [comment]
I was getting a similar issue. It was showing invalid JWT error even on admin login, so I went ahead and deleted the rsa keys in bwdata directory and restarted BW. This fixed it for me.

**Reply by kevinm416** [reply]
This also fixed it for me after copy/pasting the data folder from an existing install.

**Comment by ticmaisdev** [comment]
**for future reference:** I was able to auth (200 OK), but requests to /api would get 401 because I've got haproxy basic auth and this acl rule was being applied to /api routes. I've just disabled auth for /api path and now it works

**Reply by sorinvasilescu** [reply]
Great catch, I was using an ngxin-proxy-manager with a basic auth and that was also interfering with the login. I switched to IP filtering and everything is fine now.
